packages/core/src/hooks/useEthers.ts

```diff
@@ -14,9 +14,16 @@ export type Web3Ethers = ReturnType<typeof useWeb3React> & {
 export function useEthers(): Web3Ethers {
   const result = useWeb3React<Web3Provider>()
   const { supportedChains } = useConfig()
-  const activateBrowserWallet = useMemo(() => {
-    const injected = new InjectedConnector({ supportedChainIds: supportedChains })
-    return () => result.activate(injected)
-  }, [supportedChains])
+  const activateBrowserWallet = useCallback(
+    async (onError?: (error: Error) => void, throwErrors?: boolean) => {
+      const injected = new InjectedConnector({ supportedChainIds: supportedChains })
+      if (onError instanceof Function) {
+        await result.activate(injected, onError, throwErrors)
+      } else {
+        await result.activate(injected, undefined, throwErrors)
+      }
```

Comment:
onError needs to be checked if its a function so that acitvateBrowserWallet can be passed to onClick like
```ts
<button onClick={avticateBrowserWallet} />
```